libraries/apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/HttpCacheExtensions.kt

```diff
@@ -72,16 +73,19 @@ fun ApolloClient.Builder.httpCache(
       directory = directory,
       maxSize = maxSize,
   )
-  var cacheKey: String? = null
+  val apolloRequestToCacheKey = mutableMapOf<String, String>()
+  var apolloRequestUuid: Uuid? = null
```

Comment:
This is still racy. This code can be accessed from multiple threads. `requestUuid` needs to be passed down to the interceptor and `apolloRequestToCacheKey` protected by a lock